agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileLoader.java

```diff
@@ -108,6 +108,8 @@ PersistedFile loadTelemetriesFromDisk() {
       readFully(fileInputStream, ikeyBytes, 36);
       instrumentationKey = new String(ikeyBytes, UTF_8);
       if (!isInstrumentationKeyValid(instrumentationKey)) {
+        fileInputStream
+            .close(); // need to close FileInputStream before delete; otherwise, delete will fail.
```

Comment:
can you add a test that would have caught this?

Reply:
added.